### PR TITLE
Add area of interest mask layer

### PR DESCRIPTION
**Pull request by kdeloach**
This adds a shadow to the map with a cutout of the selected area of
interest.

The conformMapToShape logic is no longer needed since we now return
properly serialized MultiPolygon geojson.

Connects #222

### CODE REVIEW

File: src/mmw/js/src/app.js [hunk 1]
```diff
@@ -62,12 +62,12 @@ var App = new Marionette.Application({
 
 function RestAPI() {
     return {
-        getPredefinedShapes: function() {
+        getPredefinedShapes: _.memoize(function() {
```
Comment (kdeloach) [inline src/mmw/js/src/app.js]: I didn't mean to commit this but it shouldn't hurt.

File: src/mmw/js/src/core/views.js [hunk 2]
```diff
@@ -167,12 +167,65 @@ var MapView = Marionette.ItemView.extend({
...
+        if (areaOfInterest) {
+            var layer = new L.GeoJSON(areaOfInterest);
+            this._leafletMap.fitBounds(layer.getBounds(), { reset: true });
         }
     }
 });
 
+// Apply a mask over the entire map excluding bounds/shape specified.
+function applyMask(featureGroup, shapeLayer) {
+    var worldBounds = L.latLngBounds([-90, -360], [90, 360]),
```
Comment (jamesmcclain) [inline src/mmw/js/src/core/views.js]: To the best of my knowledge, latitude ranges from -180 to +180 degrees.  Is there some special reason for the double wrap?

**Comment by jamesmcclain** [comment]
You create the shadow effect by making a transparent sheet with the shape cut out of it.  This is what I see when I zoom all the way out:
screenshot from 2015-05-26 17 25 14

**Comment by jamesmcclain** [comment]
Other than the question about the double-wrap of the glob and the issue just above, it looks good to me.  I will reassign to you, for now.  Please feel free to assign back to me if/when you want to.

File: src/mmw/js/src/core/views.js (hunk 2 above)
Comment (kdeloach) [inline src/mmw/js/src/core/views.js]: I believe this is so the shadow covers more of the planet when you are zoomed really far out.

**Comment by mmcfarland** [comment]
I think we'll put some reasonable map bounds on at some point, since this is designed only for the conterminous US.  The wrap effect shouldn't be a problem.

**Comment by mmcfarland** [comment]
+1 Tested.

**Pull request merged**